sdk/keyvault/azure-keyvault-certificates/azure/keyvault/certificates/_shared/async_challenge_auth_policy.py

```diff
@@ -15,65 +15,59 @@
 """
 from typing import TYPE_CHECKING
 
-from azure.core.pipeline.policies import AsyncHTTPPolicy
+from azure.core.pipeline.policies import AsyncBearerTokenCredentialPolicy
 
-from . import HttpChallengeCache
-from .challenge_auth_policy import _enforce_tls, _get_challenge_request, _update_challenge, ChallengeAuthPolicyBase
+from . import http_challenge_cache as ChallengeCache
+from .challenge_auth_policy import _enforce_tls, _update_challenge
 
 if TYPE_CHECKING:
-    from typing import Any
-    from azure.core.credentials_async import AsyncTokenCredential
-    from azure.core.pipeline import PipelineRequest
-    from azure.core.pipeline.transport import AsyncHttpResponse
-    from . import HttpChallenge
+    from azure.core.pipeline import PipelineRequest, PipelineResponse
+    from .http_challenge import HttpChallenge
 
 
-class AsyncChallengeAuthPolicy(ChallengeAuthPolicyBase, AsyncHTTPPolicy):
+class AsyncChallengeAuthPolicy(AsyncBearerTokenCredentialPolicy):
     """policy for handling HTTP authentication challenges"""
 
-    def __init__(self, credential: "AsyncTokenCredential", **kwargs: "Any") -> None:
-        self._credential = credential
-        super(AsyncChallengeAuthPolicy, self).__init__(**kwargs)
-
-    async def send(self, request: "PipelineRequest") -> "AsyncHttpResponse":
+    async def on_request(self, request: "PipelineRequest") -> None:
         _enforce_tls(request)
+        challenge = ChallengeCache.get_challenge_for_url(request.http_request.url)
+        if challenge:
+            # Note that if the vault has moved to a new tenant since our last request for it, this request will fail.
+            await self._handle_challenge(request, challenge)
```

Comment:
It's odd to have `_handle_challenge` and `on_challenge`, and more so when neither calls the other. Consider inlining `_handle_challenge` here.